DetectorDescription/DDCMS/plugins/DDTestDumpGeometry.cc

```diff
@@ -16,35 +17,40 @@
 
 using namespace std;
 using namespace cms;
+using namespace edm;
 
-class DDTestDumpGeometry : public edm::one::EDAnalyzer<> {
+class DDTestDumpGeometry : public one::EDAnalyzer<> {
 public:
-  explicit DDTestDumpGeometry(const edm::ParameterSet&);
+  explicit DDTestDumpGeometry(const ParameterSet&);
 
   void beginJob() override {}
-  void analyze(edm::Event const& iEvent, edm::EventSetup const&) override;
+  void analyze(Event const& iEvent, EventSetup const&) override;
   void endJob() override {}
+
+private:  
+  const ESInputTag m_tag;
 };
 
-DDTestDumpGeometry::DDTestDumpGeometry(const edm::ParameterSet& iConfig)
+DDTestDumpGeometry::DDTestDumpGeometry(const ParameterSet& iConfig)
+  : m_tag(iConfig.getParameter<ESInputTag>("DDDetector"))
 {}
 
 void
-DDTestDumpGeometry::analyze(const edm::Event&, const edm::EventSetup& iEventSetup)
+DDTestDumpGeometry::analyze(const Event&, const EventSetup& iEventSetup)
 {
-  cout << "DDTestDumpGeometry::analyze:\n";
-  edm::ESTransientHandle<DDDetector> det;
-  iEventSetup.get<DetectorDescriptionRcd>().get(det);
+  LogVerbatim("Geometry") << "DDTestDumpGeometry::analyze: " << m_tag;
+  ESTransientHandle<DDDetector> det;
+  iEventSetup.get<DetectorDescriptionRcd>().get(m_tag.module(), det);
 
-  TGeoManager& geom = det->description->manager();
+  TGeoManager& geom = det->description()->manager();
```

Comment:
Can you declare this 
```suggestion
  TGeoManager const& geom = det->description()->manager();
```
and it still works?